lexers/lisp.lua

```diff
@@ -35,7 +35,12 @@ local word = lexer.alpha * (lexer.alnum + S('_-'))^0
 lex:add_rule('identifier', token(lexer.IDENTIFIER, word))
 
 -- Strings.
-lex:add_rule('string', token(lexer.STRING, "'" * word + lexer.range('"') + '#\\' * lexer.any))
+local character
+  = (P'backspace' + P'linefeed' + P'newline' + P'page'
```

Comment:
Are these case-insensitive? Some basic research (https://lisp-docs.github.io/cl-language-reference/chap-13/bd-c-dictionary/characterp_function) seems to indicate `#\Newline` and `#\Rubout` are used, rather than `#\newline` and `#rubout`.

Reply:
It seems whole Lisp is more or less case-insensitive https://www.cliki.net/Case%20sensitivity … there are some discussion what is behind that, but for the poor lexer it seems irrelevant.

Reply:
All the documentation I've found uses title-case for these escaped characters, so in order to minimize confusion, I would either:

1. Change them to use title-case.
2. Use `lexer.word_match({...}, true)` to make them case-insensitive.

Reply:
OK, I will do the second one.

Reply:
Actually, this PR was not meant as a complete modernization of `lisp.lua`, just trying to collect this one change. However, let me modernize the whole thing.